core/description/unitresources.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package description
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+)
+
+// UnitResource represents the revision of a resource used by a unit.
+type UnitResource interface {
+	Name() string
+	Revision() int
+}
+
+// UnitResourceArgs is an argument struct used to specify the revision
+// of a resource used by a unit.
+type UnitResourceArgs struct {
+	Name     string
+	Revision int
+}
+
+func newUnitResource(args UnitResourceArgs) *unitResource {
+	return &unitResource{
+		Name_:     args.Name,
+		Revision_: args.Revision,
+	}
+}
+
+type unitResources struct {
+	Version    int             `yaml:"version"`
+	Resources_ []*unitResource `yaml:"resources"`
```

Comment:
Actually, Tim hasn't done similar structs with omitempty so I might leave this as-is for consistency.